app/models/blocked_ip.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class BlockedIp < ApplicationRecord
+  belongs_to :user
+  validates_uniqueness_of :ip
+  validate :ip_format_valid
+
+  private
+
+  def ip_format_valid
+    return if IPAddr.new( ip ) rescue nil
+
+    errors.add( :ip, "IP is not a valid IP address" )
```

Comment:
Pretty minor, but this error reads a little strangely because it contains the attribute name in it, and the way we are displaying it the attribute will already be displayed (sorry - I know I included this in a previous comment): `Ip IP is not a valid IP address`. It would read better as something like `errors.add( :ip, "must be valid IP address" )`.

Another really small text change would be to tell ActiveRecord that the attribute `ip` should be displayed capitalized. This can be done by adding a translation entry for it to `activerecord.attributes`, e.g.:

```diff
diff --git a/config/locales/en.yml b/config/locales/en.yml
index 25419b5d9..ee1de85c7 100644
--- a/config/locales/en.yml
+++ b/config/locales/en.yml
@@ -9623,6 +9623,8 @@ en:
       assessment_section:
         title: Title
         body: Body
+      blocked_ip:
+        ip: IP
       blocked_user:
         # The user who has been blocked
         blocked_user: Blocked user
```